spec/data-model/README.md

```diff
@@ -22,39 +22,43 @@ Two equivalent definitions of the data model are also provided:
 
 ## Messages
 
-A `SelectMessage` corresponds to a syntax message that includes _selectors_.
-A message without _selectors_ and with a single _pattern_ is represented by a `PatternMessage`.
+A `SimpleMessage` corresponds to a syntax message that includes _selectors_.
+A message without _selectors_ and with a single _pattern_ is represented by a `ComplexMessage`.
 
 In the syntax,
-a `PatternMessage` may be represented either as a _simple message_ or as a _complex message_,
+a `ComplexMessage` may be represented either as a _simple message_ or as a _complex message_,
 depending on whether it has declarations and if its `pattern` is allowed in a _simple message_.
 
 ```ts
-type Message = PatternMessage | SelectMessage;
+type Message = SimpleMessage | ComplexMessage;
 
-interface PatternMessage {
-  type: "message";
-  declarations: Declaration[];
+interface SimpleMessage {
+  type: "simple-message";
   pattern: Pattern;
 }
 
-interface SelectMessage {
-  type: "select";
-  declarations: Declaration[];
+interface Matcher {
   selectors: Expression[];
   variants: Variant[];
 }
+
+interface ComplexMessage {
+  type: "complex-message";
+  declarations?: Declaration[];
+  body: Pattern | Matcher;
+}
 ```
 
 Each message _declaration_ is represented by a `Declaration`,
 which connects the `name` of a _variable_
 with its _expression_ `value`.
-The `name` does not include the initial `$` of the _variable_.
+The `name` does not include the `$` sigil of the _variable_,
+or the starting and ending `|` of the literals.
```

Comment:
I'm confused about this -- since this part is describing what appears on the left-hand side of a declaration, where do literals come in?